felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -523,6 +578,24 @@ func (m *bpfEndpointManager) onWorkloadEndpointUpdate(msg *proto.WorkloadEndpoin
 	wl := msg.Endpoint
 	m.allWEPs[wlID] = wl
 	m.addWEPToIndexes(wlID, wl)
+
+	// Make sure that we know the ifindex. The wep could have been removed, but we may not
+	// get a new status update for the device as the device does not change.
+	if _, ok := m.ifaceNameToIdx[wl.Name]; !ok {
+		nl, err := netlinkshim.NewRealNetlink()
```

Comment:
Best to defer dataplane operations to the CompleteDeferredWork stage if possible.